git-clone/main.tf

```
@@ -26,19 +26,46 @@ variable "agent_id" {
 }
 
 locals {
-  clone_path = var.base_dir != "" ? join("/", [var.base_dir, replace(basename(var.url), ".git", "")]) : join("/", ["~", replace(basename(var.url), ".git", "")])
+  # Remove query parameters and branch name from the URL
+  url = replace(replace(var.url, "/\\?.*/", ""), "/#.*", "")
+  # Remove tree and branch name from the URL
+  clone_url = replace(replace(local.url, "//-/tree/.*/", ""), "//tree/.*/", "")
```

Should the `/` be escaped? i.e.:
```suggestion
  clone_url = replace(replace(local.url, "/\/-\/tree\/.*/", ""), "/\/tree\/.*/", "")
```

---

I read through https://developer.hashicorp.com/terraform/language/functions/replace and I think this is a case where the Terraform syntax is just strange. It might behave weird if we start escaping the `/` inside since it doesn't have the same concept as typical regex that might look like `s/find/replace/`.

---

@mafredri  - luckily i have added some test cases to make sure changes are safe :)